pkg/openstack/cinder.go

```diff
@@ -17,17 +17,19 @@ import (
 
 // ReconcileCinder -
 func ReconcileCinder(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
-	if !instance.Spec.Cinder.Enabled {
-		return ctrl.Result{}, nil
-	}
-
 	cinder := &cinderv1.Cinder{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "cinder",
 			Namespace: instance.Namespace,
 		},
 	}
 
+	if !instance.Spec.Cinder.Enabled {
+		if res, err := CheckDeleteDisabled(ctx, helper, cinder); err != nil {
+			return res, err
+		}
+	}
```

Comment:
If the service is disabled then we delete it at L28 but then here we continue and reach the CreateOrPatch call at L34 that will recreate the service. I think we either have a service enabled and then proceed to L34 or disabled and disabled and only do `CheckDeleteDisabled` but not proceed to L34.

Reply:
Looks like we need to return if there is no error.  But strangely this works fine with my testing with no re-creation of service.  I'll recheck it.